pkg/controllers/subnet/subnetbinding_handler.go

```diff
@@ -0,0 +1,108 @@
+)
+
+var (
+	removeFinalizerDelay = 1 * time.Second
+)
+
+func requeueSubnetByBindingMapUpdate(ctx context.Context, c client.Client, objOld client.Object, objNew client.Object, q workqueue.TypedRateLimitingInterface[reconcile.Request]) {
+	bindingMap := objNew.(*v1alpha1.SubnetConnectionBindingMap)
+	needFinalizer := common.IsObjectReady(bindingMap.Status.Conditions)
+	enqueueSubnets(ctx, c, bindingMap, needFinalizer, q)
+
+	oldBM := objOld.(*v1alpha1.SubnetConnectionBindingMap)
+	if common.IsObjectReady(oldBM.Status.Conditions) && oldBM.Spec.TargetSubnetName != "" &&
+		oldBM.Spec.TargetSubnetName != bindingMap.Spec.TargetSubnetName {
+		// Remove the finalizer from old target Subnet if it is not used.
+		err := enqueue(ctx, c, bindingMap.Namespace, oldBM.Spec.TargetSubnetName, false, q)
+		if err != nil {
+			log.Error(err, "Failed to requeue Subnet", "Namespace", bindingMap.Namespace, "Name", oldBM.Spec.SubnetName)
+			return
+		}
+	}
+}
+
+func enqueueSubnets(ctx context.Context, c client.Client, bindingMap *v1alpha1.SubnetConnectionBindingMap, needFinalizer bool, q workqueue.TypedRateLimitingInterface[reconcile.Request]) {
+	err := enqueue(ctx, c, bindingMap.Namespace, bindingMap.Spec.SubnetName, needFinalizer, q)
+	if err != nil {
+		log.Error(err, "Failed to requeue Subnet", "Namespace", bindingMap.Namespace, "Name", bindingMap.Spec.SubnetName)
+		return
+	}
+
+	if bindingMap.Spec.TargetSubnetName == "" {
+		return
+	}
+
+	err = enqueue(ctx, c, bindingMap.Namespace, bindingMap.Spec.TargetSubnetName, needFinalizer, q)
+	if err != nil {
+		log.Error(err, "Failed to requeue Subnet", "Namespace", bindingMap.Namespace, "Name", bindingMap.Spec.TargetSubnetName)
+		return
+	}
+}
+
+func enqueue(ctx context.Context, c client.Client, namespace, name string, needFinalizer bool, q workqueue.TypedRateLimitingInterface[reconcile.Request]) error {
+	subnetCR := &v1alpha1.Subnet{}
+	subnetKey := types.NamespacedName{Namespace: namespace, Name: name}
+	err := c.Get(ctx, subnetKey, subnetCR)
+	if err != nil {
+		log.Error(err, "Failed to get Subnet CR", "key", subnetKey.String())
+		return err
+	}
+	addedFinalizer := controllerutil.ContainsFinalizer(subnetCR, servicecommon.SubnetFinalizerName)
+	if addedFinalizer != needFinalizer {
+		log.V(1).Info("Requeue subnet", "key", subnetKey.String())
+		req := reconcile.Request{
+			NamespacedName: types.NamespacedName{
+				Name:      name,
+				Namespace: namespace,
+			},
+		}
+		if needFinalizer {
+			q.Add(req)
+		} else {
+			// Enqueue the Subnet request with 1s delay with the SubnetConnectionBindingMap deletion event.
+			// This is because when Subnet controller watches the deletion event at the same time as
+			// SubnetConnectionBindingMap Controller, and it may cause a wrong result of the Subnet dependency
+			// calculation. So we leave 1s to ensure the Subnet Controller reconciles Subnet status after
+			// the binding calculation is completed.
+			q.AddAfter(req, removeFinalizerDelay)
```

Comment:
Removed this delay in the latest change as I update to create/delete finalizers on Subnet/SubnetSet according to the existence of SubnetConnectionBindingMap CR rather than the NSX resources.